headerfs/store.go

```diff
@@ -349,9 +349,15 @@ func (h *blockHeaderStore) RollbackLastBlock() (*BlockStamp, error) {
 		return nil, err
 	}
 
+	// prevHeader, err := h.readHeader(chainTipHeight - 1)
+	// if err != nil {
+	// 	return nil, err
+	// }
+
 	return &BlockStamp{
 		Height: int32(chainTipHeight) - 1,
 		Hash:   prevHeaderHash,
+		// Timestamp: prevHeader.Timestamp,
```

Comment:
I am less certain about this one.

Reply:
Probably can remove reading the current header completely:

```
	// With this height obtained, we'll use it to read the previous header
	// from disk, so we can populate our return value which requires the
	// prev header hash.
	prevHeader, err := h.readHeader(chainTipHeight - 1)
	if err != nil {
		return nil, err
	}

	prevHeaderHash := prevHeader.BlockHash()

	// Now that we have the information we need to return from this
	// function, we can now truncate the header file, and then use the hash
	// of the prevHeader to set the proper index chain tip.
	if err := h.singleTruncate(); err != nil {
		return nil, err
	}
	if err := h.truncateIndex(&prevHeaderHash, true); err != nil {
		return nil, err
	}

	return &BlockStamp{
		Height:    int32(chainTipHeight) - 1,
		Hash:      prevHeaderHash,
		Timestamp: prevHeader.Timestamp,
	}, nil
```

Reply:
Oh yeah, `bestHeader` was only retrieved for `PrevBlock` hash.  I'll make this change too.